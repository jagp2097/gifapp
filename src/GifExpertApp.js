import React, { useState } from "react";
import { Fragment } from "react/cjs/react.production.min";
import AddCategory from "./components/AddCategory";
import GifGrid from "./components/GifGrid";

const GifExpertApp = ({ defaultCategories = [] }) => {
  // const categories = ['One Punch', 'Ratma 1/2', 'Dragon Ball', 'Captain Tsubasa'];
  // const [ categories, setCategories ] = useState(['Captain Comando']);
  const [ categories, setCategories ] = useState(defaultCategories);

  // const handleAdd = () => {
  //   setCategories([...categories, 'oso']);
  // };

  return(
    <Fragment>
      <h2>GifExpertApp</h2>
      <AddCategory 
        setCategories={ setCategories } />
      <hr />

      <ol>
        { 
          categories.map( category => ( 
            <GifGrid 
              key={ category }
              category={ category } /> 
          ))
        }
      </ol>

    </Fragment>

  );

};


export default GifExpertApp;
