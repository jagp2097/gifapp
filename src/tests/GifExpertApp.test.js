import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('Pruebas al componente <GifExpertApp>', () => {

  test('Mostrar el componente correctamente', () => {

    const wrapper = shallow(<GifExpertApp/>);

    expect(wrapper).toMatchSnapshot();

  });

  test('debe de mostrar una lista de categorias', () => {

    const categories = ['Dragon Ball', 'Captain Tsubasa'];

    const wrapper = shallow(<GifExpertApp defaultCategories = {categories} />)

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GifGrid').length).toBe(categories.length);

  });

});


