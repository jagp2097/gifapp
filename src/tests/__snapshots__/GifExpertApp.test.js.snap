// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al componente <GifExpertApp> Mostrar el componente correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas al componente <GifExpertApp> debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
    <GifGrid
      category="Captain Tsubasa"
      key="Captain Tsubasa"
    />
  </ol>
</Fragment>
`;
